### PR TITLE
cleanup detection, fix #54

### LINKED ISSUE #54: Cleanup detection
remove useless detection file

### PR DESCRIPTION
**Pull request by nicolas-rabault**

### CODE REVIEW

File: Robus/src/msgAlloc.c [hunk 1]
```diff
@@ -14,33 +14,29 @@
...
  *               |   |
- * Alloc_tasks   A   B  msg_pre_tasks        msg_tasks         Luos_tasks
- *  +-------+    |   |   +---------+        +---------+        +---------+
- *  |Alloc A|<---+   +-->|Pre msg B|---C--->|  Msg C  |---D-+->| Task D1 |
- *  |Alloc B|<-------+   |---------|<id     |---------|<id  +->| Task D2 |
- *  |-------|<id         |---------|        |---------|        |---------|<id
- *  |-------|            |---------|        |---------|        |---------|
- *  +-------+            +---------+        +---------+        +---------+
+ *               A   B    msg_tasks          Luos_tasks
+ *               |   |   +---------+        +---------+
```
Comment (nicolas-rabault) [inline Robus/src/msgAlloc.c:18]: Just remove the `|` under A

File: Robus/src/msgAlloc.c [hunk 2]
```diff
@@ -14,33 +14,29 @@
...
+ *                       +---------+        +---------+
  *
  *  - Event A : This event is called by IT and represent the end of reception of
  *              the header. In this event we get the size of the complete message
- *              so we can check if no validated tasks (msg_tasks or Luos_tasks)
- *              use the memory space needed to receive the complete message.
- *              To do it outside of IT we create an "Alloc A" task. This task
- *              will be executed in alloc_loop.
+ *              so we can check if we are at the end of the msg_buffer and report
+ *              writing pointer to the begin of meg_buffer
```
Comment (nicolas-rabault) [inline Robus/src/msgAlloc.c:28]: not meg_buffer but msg_buffer

File: Robus/src/robus.c [hunk 3]
```diff
@@ -364,7 +368,7 @@ static error_return_t Robus_MsgHandler(msg_t *input)
     uint32_t baudrate;
     msg_t output_msg;
     node_bootstrap_t node_bootstrap;
-    vm_t *vm = Recep_GetConcernedVm(&input->header);
+    ll_container_t *ll_container = Recep_GetConcernedContainer(&input->header);
```
Comment (nicolas-rabault) [inline Robus/src/robus.c:371]: Watch out container is a Luos thing this is ll_container

**Review by nicolas-rabault — COMMENTED** [review]
some cleanup to do

**Review by Salem-Luos — COMMENTED** [review]
comment fix need to be review after cleanup commit

File: Robus/inc/reception.h [hunk 4]
```diff
@@ -19,9 +19,10 @@ typedef struct __attribute__((__packed__))
     {
         struct __attribute__((__packed__))
         {
-            unsigned char identifier : 4;
-            unsigned char rx_error : 1;
-            unsigned char rx_timeout : 1;
+            uint8_t identifier : 4;
+            uint8_t rx_error : 1;
+            uint8_t rx_timeout : 1;
+            uint8_t rx_framing_error : 1;
```
Comment (nicolas-rabault) [inline Robus/inc/reception.h:25]: Watch out this is not the bug but an RX_framing_error allowing to prevent collision errors.

Comment (nicolas-rabault) [reply Robus/inc/reception.h:25]: And by the way I think there is multiple bugs on this PR you should perhaps put `bug` into `bugs` on the commit msg.

**Review by nicolas-rabault — COMMENTED** [review]
I think you miss your rebase here. There is a lot of commit changed on your branch that already are on Master...
The only positive way of doing it is to remove commits fb3f89cf4f66522ff3dbbff924c0e8bbfd52e318 and cb88743f2806f17d877dac7bf34db8f518067b8a

**Pull request merged**